app/Http/Controllers/Secretariat/UserController.php

```diff
@@ -296,6 +297,28 @@ public function uploadLanguageExam(Request $request, User $user)
         return redirect()->back()->with('message', __('general.successful_modification'));
     }
 
+    /**
+     * Remove a language exam for the user.
+     * @param Request $request
+     * @param User $user
+     * @param LanguageExam $exam
+     * @return RedirectResponse
+     */
+    public function deleteLanguageExam(Request $request, User $user, LanguageExam $exam)
+    {
+        $this->authorize('view', $user);
+        if ($exam->educationalInformation->user->id != $user->id) {
+            abort(400, 'The language exam does not belong to the given user.');
+        }
```

Comment:
Not a PHP expert, feel free to disregard. But comparing database keys seems a bit fragile. Maybe something like

```suggestion
        if ($exam->educationalInformation->user->isNot($user)) {
            abort(400, 'The language exam does not belong to the given user.');
        }
```

Reply:
Nice catch:) I've not heard about this before.